lib.php

```diff
@@ -4,30 +4,25 @@
 defined('MOODLE_INTERNAL') || die();
 class enrol_payment_plugin extends enrol_plugin {
 
-    public function enrol_page_hook(stdClass $instance){  
-        global $CFG, $USER, $OUTPUT, $PAGE, $DB;
+     public function enrol_page_hook(stdClass $instance){  
+         global $CFG, $USER, $OUTPUT, $PAGE, $DB;
 
-        //saves everything about to happen
         ob_start();
```

Comment:
Dont use output buffering here. Instead write all HTML into a string, which you keep appending to, then echo the string at the end